index.mjs

```diff
@@ -80,15 +81,20 @@ export function $(pieces, ...args) {
       printCmd(cmd)
     }
 
-    let child = $.spawn(prefix + cmd, {
+    let child = spawn(prefix + cmd, {
       cwd: process.cwd(),
       shell: typeof shell === 'string' ? shell : true,
       stdio: [promise._inheritStdin ? 'inherit' : 'pipe', 'pipe', 'pipe'],
       windowsHide: true,
-      maxBuffer: 200 * 1024 * 1024, // 200 MiB
+      maxBuffer,
     })
 
+    if (timeout) {
+      promise._timeout = setTimeout(() => promise.kill(killSignal), timeout)
```

Comment:
I'm not sure it should go into the library. It's really easy to implement in client code:
```js
let p = $`...`
setTimeout(() => p.kill('SIGINT').catch(_=>_), 1000)
await p
```

Reply:
Aha, it's possible to control timeout from a script, but it seems more convenient to hide this logic in API inners just like `cp.spawn({timeout: 1000})` does.

Reply:
Let’s think more of zx’s timeout api. In CP timeout is per execution. And here it will be globally.